src/main/webapp/3Dscript.js

```diff
@@ -194,8 +330,46 @@ function gui() {
   gui.add(poseTransform, 'scale', .5, 2, .25)
       .onChange(plotOrientation)
       .onFinishChange(plotTrajectory).name('Pose Scale Multiplier');
+
+  const max= pose.length;
+  oldEnd=max;
+  gui.add(poseStart, 'start', 0, max, max/100+1)
+      .onFinishChange(hideOrientationStart);
+  gui.add(poseEnd, 'end', 0, max, max/100+1)
+      .onFinishChange(hideOrientationEnd);
+  /* Time value. */
+  const timeStart = {time: ''};
+  time = gui.add(timeStart, '');
+
+  /* Lat value. */
+  const latStart = {lat: ''};
+  lat = gui.add(latStart, 'lat');
+
+  /* Long value. */
+  const longStart = {long: ''};
+  long = gui.add(longStart, 'long');
+
+  /* Alt value. */
+  const altStart = {alt: ''};
+  alt = gui.add(altStart, 'alt');
+
+  /* Yaw value. */
+  const yawStart = {yaw: ''};
+  yaw = gui.add(yawStart, 'yaw');
+
+  /* Pitch value. */
+  const pitchStart = {pitch: ''};
+  pitch = gui.add(pitchStart, 'pitch');
+
+  /* Roll value. */
+  const rollStart = {roll: ''};
+  roll = gui.add(rollStart, 'roll');
 }
 
+/* Global variables for raycasting. */
+const raycaster = new THREE.Raycaster();
+const mouse = new THREE.Vector2();
```

Comment:
Let's keep all the global variables at the top of the file so they're easier to find.